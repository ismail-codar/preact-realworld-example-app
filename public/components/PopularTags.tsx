import { useEffect, useState } from 'preact/hooks';

import { LoadingIndicator } from './LoadingIndicator';
import { apiGetAllTags } from '../services/api/tags';

interface PopularTagsProps {
	onClick: (tag: string) => void;
}

export function PopularTags(props: PopularTagsProps) {
	const [tags, setTags] = useState<string[]>([]);
	const [loading, setLoading] = useState(false);

	useEffect(() => {
		(async function getAllTags() {
			setLoading(true);
			setTags(await apiGetAllTags());
			setLoading(false);
		})();
	}, []);

	return (
		<div class="sidebar">
			<p>Popular Tags</p>
			<div class="tag-list">
				<LoadingIndicator show={loading} width="1em" />
				{tags.map(tag => (
					<a key={tag} href="#" class="tag-pill tag-default" onClick={() => props.onClick(tag)}>
						{tag}
					</a>
				))}
			</div>
		</div>
	);
}
