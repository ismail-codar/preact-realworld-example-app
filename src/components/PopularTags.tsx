import { FunctionalComponent, h } from 'preact'
import { Link } from 'preact-router'
import useAllTags from '../hooks/useAllTags'

const PopularTags: FunctionalComponent = () => {
  const { tags } = useAllTags()

  return (
    <div className="sidebar">
      <p>Popular Tags</p>

      <div className="tag-list">
        {tags.map(tag => (
          <Link key={tag} href={`/tag/${tag}`} className="tag-pill tag-default">{tag}</Link>
        ))}
      </div>
    </div>
  )
}

export default PopularTags
