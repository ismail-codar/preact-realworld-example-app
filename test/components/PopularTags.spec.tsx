import { h } from 'preact';
import { render, screen, waitFor } from '@testing-library/preact';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import PopularTags from '../../src/components/PopularTags';

const server = setupServer(
	rest.get(
		'https://conduit.productionready.io/api/tags',
		(_req, res, ctx) => {
			return res(
				ctx.status(200),
				ctx.json({
					tags: [
						'foo',
						'bar',
						'baz'
					]
				})
			);
		}
	)
);

test('renders the PopularTags component', () => {
	render(<PopularTags onClick={(_stringParam) => void 0} />);
	expect(
		screen.getByText('Popular Tags')
	).toBeInTheDocument();
});

test('matches snapshot', async () => {
	server.listen();

	const { asFragment } = render(<PopularTags onClick={(_stringParam) => void 0} />);
	await waitFor(() => screen.getByText('foo'));

	expect(asFragment()).toMatchSnapshot();

	server.close();
});
