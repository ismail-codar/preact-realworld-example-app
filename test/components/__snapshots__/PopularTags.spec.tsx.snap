// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <p>
      Popular Tags
    </p>
    <div
      class="tag-list"
    >
      <a
        class="tag-pill tag-default"
        href="#"
      >
        foo
      </a>
      <a
        class="tag-pill tag-default"
        href="#"
      >
        bar
      </a>
      <a
        class="tag-pill tag-default"
        href="#"
      >
        baz
      </a>
    </div>
  </div>
</DocumentFragment>
`;
